src/main/java/com/myworkspace/reportapp/repository/UserRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.myworkspace.reportapp.repository;
+
+import com.myworkspace.reportapp.entity.customer.Employee;
+import com.myworkspace.reportapp.entity.customer.Manager;
+import com.myworkspace.reportapp.entity.customer.User;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface UserRepository extends JpaRepository<User, UUID> {
```

Comment:
Since you're using inheritance with a discriminator column (`user_type`), it's usually more appropriate to query the base class User and then use casting in your query to get specific subclasses like `Employee` and `Manager`. This approach ensures that if you add more subclasses in the future, you won't need to modify your queries.

Reply:
You have right. I corrected this part of code. Prepare more elastic method for searching users